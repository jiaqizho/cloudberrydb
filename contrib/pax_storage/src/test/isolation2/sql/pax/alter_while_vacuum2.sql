-- @Description Ensures that an alter table while a vacuum operation is ok
-- 
CREATE TABLE alter_while_vacuum2_pax_column (a INT, b INT);
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;

DELETE FROM alter_while_vacuum2_pax_column WHERE a < 12000;
1: SELECT COUNT(*) FROM alter_while_vacuum2_pax_column;
2>: VACUUM alter_while_vacuum2_pax_column;
1: Alter table alter_while_vacuum2_pax_column set distributed randomly;
2<:
1: SELECT COUNT(*) FROM alter_while_vacuum2_pax_column WHERE a < 12010;
