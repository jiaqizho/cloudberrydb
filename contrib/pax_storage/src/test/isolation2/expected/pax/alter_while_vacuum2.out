-- @Description Ensures that an alter table while a vacuum operation is ok
--
CREATE TABLE alter_while_vacuum2_pax_column (a INT, b INT);
CREATE
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;
INSERT 100000
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;
INSERT 100000
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;
INSERT 100000
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;
INSERT 100000
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;
INSERT 100000
INSERT INTO alter_while_vacuum2_pax_column SELECT i as a, i as b FROM generate_series(1, 100000) AS i;
INSERT 100000

DELETE FROM alter_while_vacuum2_pax_column WHERE a < 12000;
DELETE 71994
1: SELECT COUNT(*) FROM alter_while_vacuum2_pax_column;
 count  
--------
 528006 
(1 row)
2>: VACUUM alter_while_vacuum2_pax_column;  <waiting ...>
1: Alter table alter_while_vacuum2_pax_column set distributed randomly;
ALTER
2<:  <... completed>
VACUUM
1: SELECT COUNT(*) FROM alter_while_vacuum2_pax_column WHERE a < 12010;
 count 
-------
 60    
(1 row)
